binstar_client/utils/config.py

```diff
@@ -44,14 +45,44 @@ def expand(path):
     USER_CONFIG = expand('~/.continuum/anaconda-client/config.yaml')
 
 
-# Package types used in upload/download
-PACKAGE_TYPES = {
-    'env': 'Environment',
-    'ipynb': 'Notebook',
-    'conda': 'Conda Package',
-    'pypi': 'Python Package',
+class PackageType(enum.Enum):
+    CONDA = 'conda'
+    ENV = 'env'
+    FILE = 'file'
+    NOTEBOOK = 'ipynb'
+    STANDARD_PYTHON = 'pypi'
+    STANDARD_R = 'r'
+    PROJECT = 'project'
+    INSTALLER = 'installer'
+
+    def label(self):
+        return self.get_from_mapping(PACKAGE_TYPE_LABELS, default=self.value)
+
+    def get_from_mapping(self, mapping, default=None):
+        return mapping.get(self, default)
+
+    @classmethod
+    def _missing_(cls, value):
+        for package_type_value, aliases in PACKAGE_TYPE_ALIASES.values():
```

Comment:
Inverse PACKAGE_TYPE_ALIASES looks a better option to this. A mapping like `{'standard_python': 'pypi', 'PyPI': 'pypi', 'standard_r': 'r'}` should be easier to use with a direct key access instead of a value scanning.